pkg/localstorev2/internal/chunkstore/helpers.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package chunkstore
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethersphere/bee/pkg/sharky"
+	storage "github.com/ethersphere/bee/pkg/storagev2"
+)
+
+type LocationResult struct {
+	Err      error
+	Location sharky.Location
+}
+
+// IterateLocations iterates over entire retrieval index and plucks only sharky location.
+func IterateLocations(
+	ctx context.Context,
+	st storage.Store,
+	locationResultC chan<- LocationResult,
+) {
+	go func() {
+		defer close(locationResultC)
+
+		err := st.Iterate(storage.Query{
+			Factory: func() storage.Item { return new(retrievalIndexItem) },
+		}, func(r storage.Result) (bool, error) {
+			select {
+			case <-ctx.Done():
+				return true, ctx.Err()
+			default:
+			}
+
+			entry := r.Entry.(*retrievalIndexItem)
+			locationResultC <- LocationResult{Location: entry.Location}
```

Comment:
This is naked write. Should be prevented at all costs.

We could get a context done signal while we try to return the locationResult. So your select case should have a second case for this operation and not a default case. The entry can be initialized before the select.